v2/internal/runner/runner.go

```diff
@@ -73,7 +74,7 @@ type Runner struct {
 	hostErrors        hosterrorscache.CacheInterface
 	resumeCfg         *types.ResumeCfg
 	pprofServer       *http.Server
-	customTemplates   []customTemplateRepo
+	customTemplates   *[]customtemplates.CustomTemplateProvider
```

Comment:
we can just use an array of interfaces instead of pointer to array of interface ` *[]customtemplates.CustomTemplateProvider`. 

Also, since the package name is already `customtemplates`, we can name the interface as `Provider` so it becomes more legible. A better name would be `customtemplates.Provider`

Reply:
updated